pandas/io/parsers.py

```diff
@@ -786,17 +786,25 @@ def read(self, nrows=None):
             return ret
 
         # May alter columns / col_dict
-        index, columns, col_dict = self._create_index(ret)
+        index, columns, mappings, col_dict = self._create_index(ret)
 
-        df = DataFrame(col_dict, columns=columns, index=index)
+        if not mappings:
```

Comment:
my original point was that I would not return a dict/use a dict at all. rather return a list indexed by position. you are already returning the columns, so that should be enough to figure everything out.